include/xtensor/xstrides.hpp

```diff
@@ -250,31 +292,25 @@ namespace xt
         };
     }
 
-    template <class offset_type, class S>
-    inline offset_type data_offset(const S&) noexcept
-    {
-        return offset_type(0);
-    }
-
-    template <class offset_type, class S, class Arg, class... Args>
-    inline offset_type data_offset(const S& strides, Arg arg, Args... args) noexcept
+    template <class offset_type, class S, class... Args>
+    inline offset_type data_offset(const S& strides, Args... args) noexcept
     {
-        constexpr std::size_t nargs = sizeof...(Args) + 1;
+        constexpr std::size_t nargs = sizeof...(Args);
         if (nargs == strides.size())
         {
             // Correct number of arguments: iterate
-            return static_cast<offset_type>(detail::raw_data_offset<0>(strides, arg, args...));
+            return static_cast<offset_type>(detail::raw_data_offset<0>(strides, args...));
         }
         else if (nargs > strides.size())
         {
             // Too many arguments: drop the first
-            return data_offset<offset_type, S>(strides, args...);
+            return detail::data_offset_drop<offset_type>(strides, args...);
```

Comment:
Here you could keep the recursive call if you revert the change of the function signature